src/ngraph/node.hpp

```diff
@@ -189,6 +189,8 @@ namespace ngraph
         virtual const op::AutoBroadcastSpec& get_autob() const;
         /// \returns true if the node can decompose
         virtual bool supports_decompose() const { return false; }
+        /// \returns true replaces this node with a constant
```

Comment:
@diyessi, please provide more details on what this function should do and what is the expectations (constant inputs etc.). Could you also provide explicit confirmation that replacing it directly in a graph by a constant is more solid approach in comparison to just returning a vector of Constants without doing any in-graph manipulations? Can you make sure that it won't lead to too much extra code when we just want to evaluate some node in a graph without folding it to a Constant (some kind of partial graph evaluation in addition to shape and type propagation)? For example, in MO we are propagating values through the graph without folding parts of a graph into Constants. This apprach helps us to keep shape sub-expression in a graph untouched but do shape inference for entire graph in the same time. I would like to have a solution that doesn't block or complicate this path.

Reply:
#4489 sketches partial evaluation. 
The request was for a way for custom ops to add have their own constant folders. If they also need to participate in partial evaluation, then output vector results make more sense; that wasn't part of the request.

Reply:
@diyessi, sure it was not a part of that particular request but I want to unlock the possibility to re-use it for something related.

Reply:
@diyessi, can we finalize this interface by the end of this week just to unblock others? If you think that returning a vector of constants is a better approach considering the other usages that I mentioned, please change the signature. Can you provide more examples with implementing this function for multi-port operations as I asked in another comment?

Reply:
Example using `ShapeOf`.